assessments/projects/openfga/self-assessment.md

```diff
@@ -0,0 +1,277 @@
+
+Database Migration Planning: When installing new versions of OpenFGA, database migrations can be executed in a way that minimizes downtime and ensure a smooth transition in the target system.
+
+### Goals
+
+* Simplify and standardize authorization processes, making them more consistent across various applications and systems.
+
+* Establish patterns and standards for externalized authorization.
+
+* Create architectural patterns, terminologies, and protocols that enable interoperability among different authorization systems.
+
+* Deliver an authorization service for any application component.
+
+* Enable centralized authorization decisions and permits diverse teams to implement authorization using a shared framework across various application components.
+
+### Non-Goals
+
+* Tools for management of groups/roles/permissions not inherently provided to the end-users.
+* Does not intend to serve as a comprehensive data repository for non-authorization related data.
+* Does not aim to provide a complete authentication and Access Control Solution.
+
+## Self-Assessment Use
+
+This self-assessment is created by the OpenFGA team to perform an internal analysis of the project's security. It is not intended to provide a security audit of OpenFGA, or function as an independent assessment or attestation of OpenFGA's security health.
+
+This document serves to provide OpenFGA users with an initial understanding of OpenFGA's security, where to find existing security documentation, OpenFGA plans for security, and general overview of OpenFGA security practices, both for development of OpenFGA as well as security of OpenFGA.
+
+This document provides the CNCF TAG-Security with an initial understanding of OpenFGA to assist in a joint-assessment, necessary for projects under incubation. Taken together, this document and the joint-assessment serve as a cornerstone for if and when OpenFGA seeks graduation and is preparing for a security audit.
+
+## Security Functions and Features
+
+See [Actors](#actors) and [Actions](#actions) for more detailed description of the critical actors, actions, and potential threats.
+
+OpenFGA, being an Open-Source project, allows for a more robust security implementation by following the Principle of Open Design.
+
+OpenFGA models authorization systems by providing the security features such as Role-based Access Control and Atrribute-based Access Control.
+
+OpenFGA boasts exceptional speed in processing secure authorization check call. This swift authorization mechanism not only enhances efficiency but also reinforces the security posture, assuring robust protection for applications and platforms for diverse scales.
+
+OpenFGA provides a wide variety of SDK's, as well as easy integration for new SDK's. This reduces the chance of critical vulnerabilities due to compatibility issues.
+
+### Security Relevant
+
+Applications track and point to specific versions of authorization models.
+
+## Project Compliance

```

Comment:
I'm putting my comments for future reference when the team is considering syncing with OpenFGA security self-assessment. I think this section can be more refined as PII might not be applicable in my sense. It is better to discuss with the team and finalize whether to include it or not. As the product level it should not be applicable in my understanding.